src/components/lock/LockForm.tsx

```diff
@@ -1,11 +1,139 @@
 import { BraindaoLogo3 } from '@/components/braindao-logo-3'
-import { Badge, Flex, IconButton, Text, Input, VStack } from '@chakra-ui/react'
-import React from 'react'
+import { getDollarValue } from '@/utils/LockOverviewUtils'
+import {
+  Badge,
+  Flex,
+  IconButton,
+  Text,
+  Input,
+  VStack,
+  useToast,
+} from '@chakra-ui/react'
+import React, { useState, useEffect, useCallback } from 'react'
 import { RiArrowDownLine } from 'react-icons/ri'
-
+import * as Humanize from 'humanize-plus'
+import { useErc20 } from '@/hooks/useErc20'
+import { useLock } from '@/hooks/useLock'
+import { useWaitForTransaction } from 'wagmi'
+import { useLockOverview } from '@/hooks/useLockOverview'
 import LockFormCommon from './LockFormCommon'
 
 const LockForm = () => {
+  const [iqToBeLocked, setIqToBeLocked] = useState(0)
+  const [exchangeRate, setExchangeRate] = useState(0)
+  const [trxHash, setTrxHash] = useState()
+  const [loading, setLoading] = useState(false)
+  const toast = useToast()
+  const { userTokenBalance } = useErc20()
+  const { lockIQ, increaseLockAmount } = useLock()
+  const { userTotalIQLocked } = useLockOverview()
+  const { refetch } = useWaitForTransaction({ hash: trxHash })
+
+  const verifyTrx = useCallback(async () => {
+    const timer = setInterval(() => {
+      try {
+        const checkTrx = async () => {
```

Comment:
This can be handled better, you can use https://wagmi.sh/docs/hooks/useWaitForTransaction to confirm the transaction

Reply:
I'm actually using useWaitForTransaction